src/DynamicData/Cache/ObservableCacheEx.cs

```diff
@@ -4764,6 +4791,34 @@ sortOrder switch
         return new TransformAsync<TDestination, TSource, TKey>(source, transformFactory, null, forceTransform).Run();
     }
 
+    /// <summary>
+    /// Projects each update item to a new form using the specified transform function.
+    /// This optimization works well under the assuption that passing the same item to the given filter predicate will always produce the same result, or results that are always functionally equivalent.
+    /// If either the predicate or the items itemselves have any mutable state, mutating it will result in undefined and potentially incorrect behavior of the result stream.
+    /// This operator also assumes that the transformation factory itself is relatively high-performance, as it will be invoked repeatedly, and its results are not internally cached.
+    /// </summary>
+    /// <typeparam name="TDestination">The type of collection items produced by the transformation.</typeparam>
+    /// <typeparam name="TSource">The type of collection items to be transformed.</typeparam>
+    /// <typeparam name="TKey">The type of the key values of each collection item.</typeparam>
+    /// <param name="source">The source stream of collection items to be transformed.</param>
+    /// <param name="transformFactory">The transformation to be applied to each item.</param>
+    /// <returns>A stream of collection changesets where upstream collection items are transformed by the given factory function.</returns>
+    public static IObservable<IChangeSet<TDestination, TKey>> TransformByValues<TDestination, TSource, TKey>(
```

Comment:
I wonder whether we can achieve this overload using an optimizer.   For example Sort accepts SortOptimisations which amongst other options allows the consumer to specify SortOptimisations.ComparesImmutableValuesOnly.

If something similar was used for Transform, or for Filter we could use different code paths in the background and eliminate the need to new named overloads.

Reply:
I could probably get behind that, in theory, but part of the problem is that this optimization is ONLY applicable for certain scenarios of filtering and transforming. Like, any of the overloads that take `IObservable<>` of something to control filtering flat-out cannot use this optimization.

So, as an example, if we were to define `FilterOptimizations.DontCacheItems`, we'd have to do one of a few things:

1. Only define one overload to take a `FilterOptimizations` paramter, the one where the filter predicate is just `Func<T, bool>`. This would kinda make the enum pointless, cause then we wouldn't be able to add a `FilterOptimizations` parameter to any other overloads, so any optimizations we might want for those would have to be in a separate enum.
2. Throw an exception if you try and specify `FilterOptimizations.DontCacheItems` with an overload that supports dynamic filtering, E.G. `IObservable<Func<T, bool>>`.
3. Just silently have `FilterOptimizations.DontCacheItems` do nothing for overloads that support dynamic filtering, leading users to think they're getting a performance boost when they're not.

On a separate note (and this comes with a grain of salt, because I have not done a DEEP dive into it, just seen it on the surface) my current impression of how sorting works in DynamicData is that I don't really like it. I don't like how much seemingly-unnecessary stuff there is inside `ISortedChangeSet<>`, and `SortOptimizations` is part of that. I have found that the "extra" stuff in `ISortedChangeSet<>` makes it impossible to implement some of the most common querying scenarios supported by LINQ. All that being said, I'm initially very wary about the idea of adding `FilterOptimizations` in the same manner as `SortOptimizations`.